### ISSUE TITLE
Create a file which will store lists of all modules and test modules ported to Python 3, and a test runner that utilizies it

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5869]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#5869|
|Type|task|
|Created|2012-08-15 14:22:11Z|
|Branch|https://github.com/twisted/twisted/tree/python3-list-5869|

For various reasons (installation, testing, API documentation generation), it would be good to have structured list of all modules and test modules ported to Python 3. I propose adding a module which will contain two lists, one of regular modules and one of test modules. In addition, a test runner will be added (using built-in unittest for now) to run the Python 3-ported tests.

<details><summary>Searchable metadata</summary>

```
trac-id__5869 5869
type__task task
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_python3_list_5869 branches/python3-list-5869
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1345040531000000 1345040531000000
changetime__1345224182000000 1345224182000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst commented|
|-|-|

(In [35221]) Branching to 'python3-list-5869'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @itamarst|
|-|-|

Please add documentation explaining what the presence of a package in the list means (eg "twisted.python").

> Since it will contain only data, no tests are necessary.

Ha ha ha.  Why didn't you make it a data file, then? :)

What are the immediate plans for this data?  It would be easier to evaluate whether this satisfies the requirements if there were at least one consumer of the information.

Perhaps this should include `admin/run-python3-tests` as well?

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set owner to @exarkun|
|-|-|

OK, moved file, added test runner. Unfortunately there's no actual tests to run until I ported twisted.python.versions using a couple of open tickets.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @itamarst|
|-|-|

It looks like the interface to `unittest.main` is sort of ridiculous.  I think we need to do `unittest.main(module="twisted", argv=["python3.3"] + testModules)` and then put things like "python.test.test_versions" into `testModules`.  At least, that's the only way I could get the runner to even try to do anything.

Should there be some indication of what types of strings belong in the two lists?  I guess the identifier type?  otoh this is no longer a module we really care particularly about the quality of, so long as it sorta kinda works.

Merge if you can understand the `unittest.main` thing.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_6"></a>|@itamarst commented|
|-|-|

The strings in there need to be "identifiers", yeah, but basically native string format is what both Python 2 and Python 3 expect for module names, and our module names are always ASCII-compatible anyway, so no need to really enforce it in any way.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_7"></a>|@itamarst set status to `closed`|
|-|-|

(In [35269]) Merge python3-list-5869: Python 3 test runner.

Author: itamar
Review: exarkun
Fixes: [#5869](https://github.com/twisted/twisted/issues/5869)

The runner is not actually functional until we've ported a couple more modules, but I tested it manually.
